plugins/modules/source_control/git/git_commit.py

```diff
@@ -0,0 +1,243 @@
+
+    if rc == 128:
+        module.fail_json(msg=error)
+
+    if rc == 0:
+        # no output returned when rc = 0
+        return
+
+
+def git_commit(module):
+
+    def git_commit_run(cmd, path):
+
+        rc, output, error = module.run_command(cmd, cwd=path)
+
+        if rc == 1:
+            module.fail_json(msg=error)
+        if rc == 0:
+            return output
+
+    result = dict(changed=False)
+    empty_commit = module.params.get('empty_commit')
+    comment = module.params.get('comment')
+    path = module.params.get('path')
+    cmd_porcelain = ['git', 'commit', '--porcelain']
+
+    # rc is always 1 for "git commit --porcelain"
+    _rc, output, error = module.run_command(cmd_porcelain, cwd=path)
+
+    if error:
+        module.fail_json(msg=error)
+
+    # do not allow empty commit with empty_commit=fail
+    if empty_commit == 'fail' and not output:
+        module.fail_json(msg='No empty commit allowed with empty_commit=fail')
+
+    # add and commit if output and empty_commit=fail
+    if empty_commit == 'fail' and output:
+
+        git_add(module)
+
+        commit_cmds = [
+            'git',
+            'commit',
+            '-m',
+            '"{0}"'.format(comment),
+        ]
+
+        output = git_commit_run(commit_cmds, path)
+
+        if output:
+            result.update(
+                git_commit=output,
+                changed=True
+            )
+
+            return result
+
+    # if empty_commit=allow skip git_add()
+    if empty_commit == 'allow' and not output:
+        commit_cmds = [
+            'git',
+            'commit',
+            '--allow-empty',
+            '-m',
+            '"{0}"'.format(comment),
+        ]
+
+        output = git_commit_run(commit_cmds, path)
+
+        if output:
+            result.update(
+                git_commit=output,
+                changed=True
+            )
+
+            return result
+
+    # if empty_commit=allow and pending stages, issue a warning and carry-on
+    if empty_commit == 'allow' and output:
+        display.warning(msg='You are going to run an empty commit but you have pending changes: {0}'.format(output))
+
+        commit_cmds = [
+            'git',
+            'commit',
+            '--allow-empty',
+            '-m',
+            '"{0}"'.format(comment),
+        ]
+
+        output = git_commit_run(commit_cmds, path)
+
+        if output:
+            result.update(
+                git_commit=output,
+                changed=True
+            )
+
+            return result
+
+    # return if empty_commit=skip and nothing to commit
+    if empty_commit == 'skip' and not output:
+
+        result.update(changed=False)
+
+        return result
+
+    # if empty_commit=skip and pending stages, issue a warning and return
+    if empty_commit == 'skip' and output:
+        display.warning('You are going to skip commit but you have pending changes: {0}'.format(output))
```

Comment:
I think this behavior is seriously wrong. The commit needs to be created, since it is not empty.